pypesto/visualize/parameters.py

```diff
@@ -1,64 +1,89 @@
 import matplotlib.pyplot as plt
+import matplotlib.axes
 from matplotlib.ticker import MaxNLocator
 import numpy as np
 import numbers
-from .reference_points import create_references
+
+from typing import Iterable, List, Optional, Sequence, Tuple, Union
+
+from ..result import Result
+from .reference_points import create_references, ReferencePoint
 from .clust_color import assign_colors
 from .clust_color import delete_nan_inf
 from .misc import process_result_list
 
 
-def parameters(results, ax=None, free_indices_only=True, lb=None, ub=None,
-               size=None, reference=None, colors=None, legends=None,
-               balance_alpha=True, start_indices=None):
+def parameters(
+        results: Union[Result, Sequence[Result]],
+        ax: Optional[matplotlib.axes.Axes] = None,
+        parameter_indices: Union[str, Sequence[int]] = 'free_only',
+        lb: Optional[Union[np.ndarray, List[float]]] = None,
+        ub: Optional[Union[np.ndarray, List[float]]] = None,
+        size: Optional[Tuple[float, float]] = None,
+        reference: Optional[List[ReferencePoint]] = None,
+        colors: Optional[Union[List[float], List[List[float]]]] = None,
+        legends: Optional[Union[str, List[str]]] = None,
+        balance_alpha: bool = True,
+        start_indices: Optional[Union[int, Iterable[int]]] = None
+) -> matplotlib.axes.Axes:
     """
     Plot parameter values.
 
     Parameters
     ----------
-    results: pypesto.Result or list
+    results:
         Optimization result obtained by 'optimize.py' or list of those
-    ax: matplotlib.Axes, optional
+    ax:
         Axes object to use.
-    free_indices_only: bool, optional
-        If True, only free parameters are shown. If
-        False, also the fixed parameters are shown.
-    lb, ub: ndarray, optional
+    parameter_indices:
+        Specifies which parameters should be plotted. Allowed string values are
+        'all' (both fixed and free parameters will be plotted)  and
+        'free_only' (only free parameters will be plotted)
```

Comment:
Cool feature. Was using hacks in PESTO back then to have precisely this functionality...